seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceSplitEnumerator.java

```diff
@@ -307,10 +307,13 @@ private Set<KafkaSourceSplit> getTopicInfo() throws ExecutionException, Interrup
         }
         log.info("Discovered topics: {}", topics);
         Collection<TopicPartition> partitions =
-                adminClient.describeTopics(topics).all().get().values().stream()
+                adminClient.describeTopics(topics).allTopicNames().get().values().stream()
                         .flatMap(
                                 t ->
                                         t.partitions().stream()
+                                                .filter(
+                                                        partitionInfo ->
+                                                                partitionInfo.leader() != null)
```

Comment:
Can we provide a warning ？

Reply:
And what should he do when the partition is restored

Reply:
> And what should he do when the partition is restored

After enabling dynamic partition discovery, the next fetch will occur once the partition is restored. If it is not enabled, data will only be written to partitions with an existing leader. I think this will not negatively impact the user.

Reply:
> Can we provide a warning ？

done.

Reply:
> > And what should he do when the partition is restored
> 
> After enabling dynamic partition discovery, the next fetch will occur once the partition is restored. If it is not enabled, data will only be written to partitions with an existing leader. I think this will not negatively impact the user.

This easily gives the impression to users that it is executing as expected. Should we also provide failure strategies, although it is not mandatory?

Reply:
> > > And what should he do when the partition is restored
> > 
> > 
> > After enabling dynamic partition discovery, the next fetch will occur once the partition is restored. If it is not enabled, data will only be written to partitions with an existing leader. I think this will not negatively impact the user.
> 
> This easily gives the impression to users that it is executing as expected. Should we also provide failure strategies, although it is not mandatory?

Can we provide users with a parameter similar to ignoreNoLeaderPartition, which defaults to false, allowing users to choose?

Reply:
If batch processing is used, this change will result in data loss. The user is not aware that a partition is unavailable.

Reply:
> If batch processing is used, this change will result in data loss. The user is not aware that a partition is unavailable.

Sorry, I don't quite understand in what situation data would be lost. Could you please explain it to me?

Reply:
Some partitions can not be read, but users doesn't know.

Reply:
> > > > And what should he do when the partition is restored
> > > 
> > > 
> > > After enabling dynamic partition discovery, the next fetch will occur once the partition is restored. If it is not enabled, data will only be written to partitions with an existing leader. I think this will not negatively impact the user.
> > 
> > 
> > This easily gives the impression to users that it is executing as expected. Should we also provide failure strategies, although it is not mandatory?
> 
> Can we provide users with a parameter similar to ignoreNoLeaderPartition, which defaults to false, allowing users to choose?

I think it can be done. When this parameter is not used, the task will fail directly and provide the corresponding prompt.

Reply:
> Can we provide users with a parameter similar to ignoreNoLeaderPartition, which defaults to false, allowing users to choose?

Ok for me.